### PR TITLE
Database | Fetch Latest Detection

### PR DESCRIPTION
**Pull request by stalberm**
## Description
* Adds method in the Detection model to retrieve the latest Detection in the database
* Adds test for the method

## Related Issues
* References #39
* References #56

## Screenhots
image

### CODE REVIEW

**Review by theusaf — APPROVED** [review]
Nice tests!

**Review by kimberlymarkley — CHANGES_REQUESTED** [review]
Your functions make this code so readable—nice work. Just a tiny ask: please adjust the imports in main.dart so that the binsight.ai package on line 4 is under Project imports and web_socket_channel on line 26 is under Package imports.

File: lib/main.dart [hunk 1]
```diff
@@ -81,22 +85,71 @@ void main() async {
     }
   }
 
-  runApp(BinsightAiApp(skipSetUp: devices.isNotEmpty));
+  runApp(BinsightAiApp(skipSetUp: true));
```
Comment (bpugh5) [inline lib/main.dart:88]: Is this hardcoded bit meant to stay in the PR?

File: lib/main.dart [hunk 2]
```diff
@@ -81,22 +85,71 @@ void main() async {
...
+    channel.sink.add(jsonEncode(subscriptionMessage));
+
+    handleMessages(channel);
+  }
+
+  @override
+  void didChangeAppLifecycleState(AppLifecycleState state) {
+    super.didChangeAppLifecycleState(state);
+    //Minimized
+    if (state == AppLifecycleState.paused) {
```
Comment (bpugh5) [inline lib/main.dart:129]: I think all this logic and stuff is super rad btw

**Review by bpugh5 — DISMISSED** [review]

**Review by bpugh5 — APPROVED** [review]

**Review by kimberlymarkley — APPROVED** [review]

**Pull request merged**